geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/AvroIngestProcessor.scala

```diff
@@ -143,41 +99,158 @@ trait AvroIngestProcessor extends FeatureTypeProcessor {
         override def process(in: InputStream): Unit = {
           val reader = new AvroDataFileReader(in)
           try {
-            val features = buildConverter(reader.getSft, sft) match {
-              case None => reader
-              case Some(converter) => reader.map(converter.apply)
+            // noinspection ScalaDeprecation
+            val nameArg = Option(file.getAttribute(SftNameAttribute)).orElse {
+              val fno = Option(context.getProperty(FeatureNameOverride).evaluateAttributeExpressions().getValue)
+              val old = Option(context.getProperty(Properties.AvroSftName).getValue)
+              if (old.isEmpty) {
+                fno
+              } else if (fno.isEmpty) {
+                logger.warn(
+                  s"Using deprecated property ${Properties.AvroSftName.getName} - " +
+                      s"use ${FeatureNameOverride.getName} instead")
+                old
+              } else {
+                logger.warn(
+                  s"Ignoring property ${Properties.AvroSftName.getName}: ${old.get} " +
+                      s"in favor of ${FeatureNameOverride.getName}: ${fno.get}")
+                fno
+              }
```

Comment:
What happens if both old and fno are empty?

Reply:
that's ok, then the nameArg will be None and it will use the name from the sft in the avro file